maths/rsa.py

```diff
@@ -23,66 +23,72 @@
 
 import random
 
-from rabin_miller import * # is_prime
-from extended_gcd import * # extended_gcd
 
-"""
-generate a prime with k bits
-"""
-def genprime(k):
-    random.seed()
-    while True:
-        n = random.randrange(int(2 ** (k - 1)),int(2 ** k))
-        if is_prime(n,128):
-            return n
-
-
-"""
-calculate the inverse of a mod m
-that is, find b such that (a * b) % m == 1
-"""
 def modinv(a, m):
-        b = 1
-        while ((a*b) % m != 1):
-            b += 1
-        return b
+    """calculate the inverse of a mod m
+    that is, find b such that (a * b) % m == 1"""
+    b = 1
+    while not (a * b) % m == 1:
+        b += 1
+    return b
+
+
+def generate_key(k, seed=None):
+    """
+    the RSA key generating algorithm
+    k is the number of bits in n
+    """
+
+    def gen_prime(k, seed=None):
```

Comment:
generates this function k-bits prime numbers?

Reply:
Its the same function that was used previously, the code is short and seems correct to me.